src/components/Youtube/YoutubeVideo.tsx

```diff
@@ -0,0 +1,68 @@
+import { useState } from "react";
+import { IoCloseSharp } from "react-icons/io5";
+import { AiOutlineReload } from "react-icons/ai";
+import { useToggleYoutube } from "@Store";
+
+export const YoutubeVideo = () => {
+  const defaultVideoId = "TYCBicKyVhs";
+  const playlistText = "?listType=playlist&list=";
+  const { isYoutubeToggled, isYoutubeShown, setIsYoutubeToggled } = useToggleYoutube();
+  const [inputText, setInputText] = useState("");
+  const [videoId, setVideoId] = useState(defaultVideoId);
+  const regex = new RegExp("[^=|\/]*$");
+
+  const handleVideoChange = (streamName: string) => {
+    const id = streamName.match(regex)[0];
+    
+    if (id.length > 0) setVideoId(id);
+
+    if (id.length > 11) setVideoId(playlistText + id);

```

Comment:
Why hard coded to 11?

Reply:
The regex above makes sure that it gets the youtube video/playlist id from the link the user provides, for an individual video it is guaranteed to be 11 characters long which is why it's hardcoded that way. However, there was an issue with it where it wouldn't properly work with videos that have a timestamp attached (ex. https://www.youtube.com/watch?v=aonx9JRry7o**&t=7s**). The new commits should fix this issue